python/cudf/cudf/tests/test_orc.py

```diff
@@ -1839,3 +1839,35 @@ def test_orc_writer_cols_as_map_type_error():
         TypeError, match="cols_as_map_type must be a list of column names."
     ):
         df.to_orc(buffer, cols_as_map_type=1)
+
+
+@pytest.mark.parametrize("engine", ["cudf", "pyarrow"])
+def test_negative_timestamp(tmpdir, engine):
+    expected = cudf.DataFrame(
+        {
+            "a": [
+                pd.Timestamp("1969-12-31 23:59:59.000123"),
+                pd.Timestamp("1969-12-31 23:59:58.000999"),
+                pd.Timestamp("1969-12-31 23:59:58.001001"),
+                pd.Timestamp("1839-12-24 03:58:56.000826"),
+            ]
+        }
+    )
+    cudf_fname = tmpdir.join("cudf_neg_ts.orc")
+    expected.to_orc(cudf_fname)
+
+    cudf_got = cudf.read_orc(cudf_fname, engine=engine)
+    assert_eq(expected, cudf_got)
+    assert_eq(expected, pd.read_orc(cudf_fname))
+    assert pyarrow.orc.ORCFile(cudf_fname).read().equals(cudf_got.to_arrow())
+
+    pyorc_fname = tmpdir.join("pyorc_neg_ts.orc")
```

Comment:
Should this part using pyorc be in a separate test than the one using only the cudf reader? (I would defer to convention -- if other tests do both together, that's fine, but it does feel like it's asserting too much for one test.)

Reply:
How about a separate case of the same test? I can have `reader` (cudf, pandas, pyarrow) and `writer` (cudf, pyorc) parameters(fixtures?) and cover all six combinations with few assertions in each test.

Reply:
Sounds good — but I’m also fine with the current state if you don’t want to make that change.

Reply:
Separated the test into a reader and writer tests and removed redundant checks.